hpx/parallel/algorithms/for_each.hpp

```diff
@@ -175,6 +192,37 @@ namespace hpx { namespace parallel { inline namespace v1
                     std::move(first));
             }
         };
+        /// Non Segmented implementation
+        template <typename ExPolicy, typename InIter, typename Size, typename F,
+            typename Proj = util::projection_identity>
+        typename util::detail::algorithm_result<ExPolicy, InIter>::type
+        for_each_n_(ExPolicy && policy, InIter first, Size count, F && f,
+            std::false_type, Proj && proj = Proj())
+        {
+            typedef std::integral_constant<bool,
+                    parallel::execution::is_sequenced_execution_policy<
+                        ExPolicy
+                    >::value ||
+                   !hpx::traits::is_forward_iterator<InIter>::value
+                > is_seq;
+            return detail::for_each_n<InIter>().call(
+                std::forward<ExPolicy>(policy), is_seq(),
+                first, std::size_t(count), std::forward<F>(f),
+                std::forward<Proj>(proj));
+        }
+        /// Segmented implementaion using for_each.
+        template <typename ExPolicy, typename InIter, typename Size, typename F,
+            typename Proj = util::projection_identity>
+        typename util::detail::algorithm_result<ExPolicy, InIter>::type
+        for_each_n_(ExPolicy && policy, InIter first, Size count, F && f,
+            std::true_type, Proj && proj = Proj())
+        {
+            auto last = first;
+            detail::advance(last,std::size_t(count));
```

Comment:
This comment applies to many spots in the code, btw. Clang-format will help with that as well (at least for the files you created).